src/systems/detachable_joint/DetachableJoint.cc

```diff
@@ -96,15 +96,103 @@ void DetachableJoint::Configure(const Entity &_entity,
   }
 
   // Setup detach topic
-  std::vector<std::string> topics;
-  if (_sdf->HasElement("topic"))
+  std::vector<std::string> detachTopics;
+  if (_sdf->HasElement("detach_topic"))
   {
-    topics.push_back(_sdf->Get<std::string>("topic"));
+    detachTopics.push_back(_sdf->Get<std::string>("detach_topic"));
   }
-  topics.push_back("/model/" + this->model.Name(_ecm) +
+  detachTopics.push_back("/model/" + this->model.Name(_ecm) +
       "/detachable_joint/detach");
-  this->topic = validTopic(topics);
+  this->detachTopic = validTopic(detachTopics);
+  igndbg << "Detach topic is: " << this->detachTopic << std::endl;
 
+  // Warn if DEPRECATED <topic> tag is used instead of <detach_topic>
+  if (_sdf->HasElement("topic"))
+  {
+    ignerr << "<topic> tag is DEPRECATED. Please use <detach_topic> instead."
+           << std::endl;
+    if (_sdf->HasElement("detach_topic"))
+    {
+      if (_sdf->Get<std::string>("topic") !=
+          _sdf->Get<std::string>("detach_topic"))
+      {
+        ignerr << "<topic> and <detach_topic> tags have different contents. "
+                  "Please verify the correct string and use <detach_topic>."
+               << std::endl;
+      }
+      else
+      {
+        ignwarn << "Ignoring <topic> tag and using <detach_topic> tag."
+                << std::endl;
+      }
+    }
+    else
+    {
+      detachTopics.pop_back();
+      detachTopics.push_back(_sdf->Get<std::string>("topic"));
+      this->detachTopic = validTopic(detachTopics);
+      ignerr << "Implicitly converted <topic> to <detach_topic>. "
+             << "Detach topic is: " << this->detachTopic << std::endl;
+    }
+  }
+
+  // Setup subscriber for detach topic
+  this->node.Subscribe(
+      this->detachTopic, &DetachableJoint::OnDetachRequest, this);
+
+  igndbg << "DetachableJoint subscribing to messages on "
+         << "[" << this->detachTopic << "]" << std::endl;
+
+  // Setup attach topic
+  std::vector<std::string> attachTopics;
+  if (_sdf->HasElement("attach_topic"))
+  {
+    attachTopics.push_back(_sdf->Get<std::string>("attach_topic"));
+  }
+  attachTopics.push_back("/model/" + this->model.Name(_ecm) +
+      "/attachable_joint/attach");
```

Comment:
Should this be `detachable_joint/attach` since `detachable_joint` is the name of the system?

Reply:
addressed [here](https://github.com/gazebosim/gz-sim/pull/1687/commits/4154296935bea0590eba0e61795cd0d0372e8e69#diff-fea76ed3ef6466553a4fa49c447f48edf452320889302ccca96cb3e2817b460fR149)